libraries/libfc/test/test_bls.cpp

```diff
@@ -294,54 +294,54 @@ BOOST_AUTO_TEST_CASE(bls_regenerate_check) try {
 BOOST_AUTO_TEST_CASE(bls_prefix_encoding_check) try {
 
   //test no_throw for correctly encoded keys
-  BOOST_CHECK_NO_THROW(bls_private_key("PVT_BLS_O+gS5yNxVtSwL0/Uhl1IVqu/Y7Gq4qdrtB66EIb192ENfK8D"));
-  BOOST_CHECK_NO_THROW(bls_public_key("PUB_BLS_dEvut0ydHevDGP6Ef3O4Iq6QXf9jUcMUT1nCJRX+JRYlFYrO/qKt/x439vUJ2DkZ32Od6AdJZ+S9dWRE9Sy+7Q6bNjpoIOP0cWzkKC1DqmhfE3paW+KThA3noLkV8SsILcfxpQ=="));
-  BOOST_CHECK_NO_THROW(bls_signature("SIG_BLS_prHR3PtNGJQLEhqcuHqj5Ty6FkNFo+ih32+ZHh6LH74+SKlTgq4PWtudoYt8heEZjPyNDRrqfRoYoTlAZ1mpW0QzgyGRXU+lfZ27M9Bg1mNS0MI6wWL4ZG9E8bchiMUVWijpX66sc11t60m/g8/vJIf1tIuFLhKCcX57OVCoXisciI7D21b3tKjb7VAlc2oNEoJx17XOafIWvcH1YKAc2uv9T/ocAlE3VQNprXKuGaZYA9Q5yzaOhVgGYxrjv/wNv0DlzA=="));
+  BOOST_CHECK_NO_THROW(bls_private_key("PVT_BLS_vh0bYgBLOLxs_h9zvYNtj20yj8UJxWeFFAtDUW2_pG44e5yc"));
+  BOOST_CHECK_NO_THROW(bls_public_key("PUB_BLS_82P3oM1u0IEv64u9i4vSzvg1-QDl4Fb2n50Mp8Sk7Fr1Tz0MJypzL39nSd5VPFgFC9WqrjopRbBm1Pf0RkP018fo1k2rXaJY7Wtzd9RKlE8PoQ6XhDm4PyZlIupQg_gOuiMhcg=="));

```

Comment:
One reason for using base64url was to avoid the `=` padding characters. However, our implementation of base64url does still add the padding characters. Worse yet, the implementation it is based on https://github.com/ReneNyffenegger/cpp-base64 latest code used `.` as a padding character for base64url which is clearly wrong. The spec https://datatracker.ietf.org/doc/html/rfc4648#section-5 actually says that `.` was not used because `The remaining unreserved URI character is ".",
   but some file system environments do not permit multiple "." in a
   filename, thus making the "." character unattractive as well.`

I would say to use my version https://github.com/heifner/base64 but it also currently has the incorrect implementation for base64url since it is based off of ReneNyffenegger version. I will fix that now that I realize it is a broken.

Reply:
Thanks. @arhag yesterday said we don't need to add padding. We can talk it at today's meeting.

Reply:
Our implementation of base64url encode adds the padding but that can be changed not to without affecting consensus. Our implementation of base64url decode, which is consensus, does not care about padding, see for example this test that deliberately removes the padding,
https://github.com/AntelopeIO/leap/blob/6d188675ac97ef00aa48fe9e64e3e73a1ada09c4/libraries/libfc/test/crypto/test_webauthn.cpp#L210-L216

So if you just change base64url encode to not add padding it seems like it'd work.

Reply:
Thanks. We only change base64url.

Reply:
That new implementation breaks consensus due to how it changes base64url decode.

The existing implementation probably just needed something simple like
```c++
while(ret.ends_with('='))
   ret.pop_back();
```

Reply:
Should we go back to original implementation. One thing about the new implementation is the same decode works on both url-safe and url-non-safe encoding; it was at first a surprise but correct.